understanding/21/label-in-name.html

```diff
@@ -9,23 +9,60 @@
 		<h1>Understanding Label in Name</h1>
 		<section id="intent">
 			<h2>Intent</h2>
-			<p>The intent of this Success Criterion (SC) is to help ensure that people with disabilities who rely on visual labels can also use
-			those labels programmatically. Controls are often labeled with visible text. Controls also have a programmatic label, known
-			as its <a href="https://www.w3.org/TR/accname-aam-1.1/">Accessible Name</a>. Users have a much better experience if the visible text labels of controls match their accessible names.</p>
-			<p>Speech input users can navigate by speaking the visible text labels of menus, links and buttons that appear on the
-			screen. It's confusing to speech input users when they say a visible text label they see, but the speech input does not work
-			because the accessible name that is enabled as a speech input command does not match the visible label.</p>
-			<p>In addition, when the accessible names of controls are visually hidden, speech users may accidentally activate these controls when they happen to speak words that map to these hidden accessible names.</p>
-			<p>Text-to-speech users will also have a better experience if the text they hear matches the text they see on the screen.</p>
-			<p>Sometimes authors want to add additional text to provide more context to a link or other control to make it more descriptive for users of assistive technologies. In this case, it is a recommended practice to add that text <strong>after</strong> the visible label so the accessible name starts with the visible label. In the case of the search button, the accessible name could be "search this site", in the case of the home link, the text could be "home - go to start page". In both cases, the hidden portion of the label may be supplied by different means, for example, as text accessibly hidden via CSS or by using the aria-label or the aria-labelledby attribute.</p>
-			<p>This issue is even more important for speech input and text-to-speech users who also have cognitive challenges. It’s an extra
-			cognitive load for a speech input user to remember and say a speech command that is different from the visible label they see on a
-			control. It’s also an extra cognitive load for a text-to-speech user to absorb and understand speech output that does not
-			match the visible label.</p>
+			<p>The intent of this Success Criterion (SC) is to make text that is <em>visually</em> associated with a component also associated with the component <em>programmatically</em>. This helps ensure that people with disabilities can rely on visible labels as a means to interact with the components.</p>
```

Comment:
To be picky, the first sentence makes it sound to me like it has to be the same piece of text in the markup that is associated. However, it can be duplicated as well (i.e. the visual text itself, and then an alternative that duplicates that particular text, e.g. `First name <input ... aria-label="First name">`(

Reply:
If you have a wording suggestion that solves that, I'll be happy, but I find trying to be clear about text vs text string, etc., has resulted in it being more confusing (or at least a lot harder to parse). Yes, one could just recreate the visible string in an aria-label; however at some point automated rules are going to start failing the multiple variations which would probably be okay with speech recognition -- e.g.,  "First Name:" and "firstname" -- and it is just so much easier for everyone if aria-label is left out where possible.

Reply:
I would tend not to handwave away the ways in which it's perfectly valid to pass this SC (e.g. using `aria-label`). So starting with something like

```
The intent of this Success Criterion (SC) is to make sure that text that is <em>visually</em> associated with a component also forms part of the component's accessible name.
```

and then adding more explanations, references, examples could be better?

Examples could be along the lines of (grabbing something I threw together not so long ago for an unofficial explainer):

```
<a href="..." aria-label="Please read more about WCAG 2.1">Read more</a> would pass, as the “Read more” string (disregarding capitalization) is present in the aria-label

<a href="..." aria-label="Read about WCAG 2.1">Read more</a> would fail, as the string is not present

<a href="..." aria-label="Read some more about WCAG 2.1">Read more</a> would fail, as the aria-label does not contain the actual string “read more” (the word “some” is between “Read” and “more”)
```

Reply:
How about:
The intent of this Success Criterion (SC) is to make the accessible name include (or match) the text used for the visual label for a component. This helps ensure that people with disabilities can rely on visible labels as a means to interact with the components.